tests/t244_report_task_field.py

```diff
@@ -4,24 +4,19 @@
 
 class TestCase(TestBase):
     def __init__(self):
-        TestBase.__init__(self, 'thread-name', """
+        TestBase.__init__(self, 'thread', """
```

Comment:
Why is this changed?

Reply:
It's because the binary generated `t-thread-name` is tricky in my `gcc-9.4.0` build.
```
$ ./runtest.py 244
Start 1 tests with 1 worker
Test case                 pg             finstrument-fu
------------------------: O0 O1 O2 O3 Os O0 O1 O2 O3 Os
244 report_task_field   : OK OK NG NG NG OK OK OK OK OK
```
The trace result of `t-thread-name.O2` looks wrong as follows.
```
$ gcc -o t-thread-name.O0 -fno-inline -pg -O0  s-thread-name.c -pthread
$ gcc -o t-thread-name.O2 -fno-inline -pg -O2  s-thread-name.c -pthread

$ uftrace --no-libcall --no-event t-thread-name.O0
# DURATION     TID     FUNCTION
            [ 47872] | main() {
            [ 47887] | thread_first() {
   5.936 us [ 47887] |   foo();
   0.207 us [ 47887] |   bar();
   8.701 us [ 47887] | } /* thread_first */
            [ 47888] | thread_second() {
   6.322 us [ 47888] |   foo();
   0.212 us [ 47888] |   bar();
   9.181 us [ 47888] | } /* thread_second */
            [ 47890] | thread_fourth() {
   9.572 us [ 47890] |   foo();
            [ 47889] | thread_third() {
   0.200 us [ 47890] |   bar();
            [ 47889] |   foo() {
  16.188 us [ 47890] | } /* thread_fourth */
   5.997 us [ 47889] |   } /* foo */
   0.210 us [ 47889] |   bar();
  12.439 us [ 47889] | } /* thread_third */
 747.764 us [ 47872] | } /* main */

$ uftrace --no-libcall --no-event t-thread-name.O2
# DURATION     TID     FUNCTION
            [ 45005] | main() {
            [ 45021] | thread_second() {
            [ 45021] |   thread_first() {
   1.350 us [ 45021] |     foo();
   0.217 us [ 45021] |     bar();
   3.462 us [ 45021] |   } /* thread_first */
   4.418 us [ 45021] | } /* thread_second */
            [ 45020] | thread_first() {
   1.319 us [ 45020] |   foo();
   0.227 us [ 45020] |   bar();
   3.951 us [ 45020] | } /* thread_first */
            [ 45022] | thread_third() {
            [ 45022] |   thread_first() {
   1.353 us [ 45022] |     foo();
   0.226 us [ 45022] |     bar();
   3.780 us [ 45022] |   } /* thread_first */
   8.777 us [ 45022] | } /* thread_third */
            [ 45023] | thread_fourth() {
            [ 45023] |   thread_first() {
   1.383 us [ 45023] |     foo();
   0.195 us [ 45023] |     bar();
   3.291 us [ 45023] |   } /* thread_first */
   4.638 us [ 45023] | } /* thread_fourth */
 918.066 us [ 45005] | } /* main */
```
The binary generated with `-O2` option is as follows.
```asm
00000000004008d0 <thread_second>:
  4008d0:   55                      push   %rbp
  4008d1:   48 89 e5                mov    %rsp,%rbp
  4008d4:   e8 97 fd ff ff          callq  400670 <mcount@plt>
  4008d9:   5d                      pop    %rbp
  4008da:   eb d4                   jmp    4008b0 <thread_first>
  4008dc:   0f 1f 40 00             nopl   0x0(%rax)

00000000004008e0 <thread_third>:
  4008e0:   55                      push   %rbp
  4008e1:   48 89 e5                mov    %rsp,%rbp
  4008e4:   e8 87 fd ff ff          callq  400670 <mcount@plt>
  4008e9:   5d                      pop    %rbp
  4008ea:   eb c4                   jmp    4008b0 <thread_first>
  4008ec:   0f 1f 40 00             nopl   0x0(%rax)

00000000004008f0 <thread_fourth>:
  4008f0:   55                      push   %rbp
  4008f1:   48 89 e5                mov    %rsp,%rbp
  4008f4:   e8 77 fd ff ff          callq  400670 <mcount@plt>
  4008f9:   5d                      pop    %rbp
  4008fa:   eb b4                   jmp    4008b0 <thread_first>
  4008fc:   0f 1f 40 00             nopl   0x0(%rax)
```
It looks incorrect because all the above functions doesn't properly return, but it jumps to `thread_first` instead. That's why I changed the test.

Reply:
Ah, ok.  Thanks for the info.